spec/dpl/providers/codedeploy_spec.rb

```diff
@@ -41,60 +41,91 @@
+      it { should create_deployment revision: github_revision }
+    end
 
-  describe 'given --repository other' do
-    before { github_revision[:gitHubLocation][:repository] = 'other' }
-    it { should create_deployment revision: github_revision }
-  end
+    describe 'given --commit_id other' do
+      before { github_revision[:gitHubLocation][:commitId] = 'other' }
+      it { should create_deployment revision: github_revision }
+    end
 
-  describe 'given --revision_type s3 --bucket bucket --key bundle.zip' do
-    before { s3_revision[:s3Location][:key] = 'bundle.zip' }
-    it { should create_deployment revision: s3_revision }
-  end
+    describe 'given --repository other' do
+      before { github_revision[:gitHubLocation][:repository] = 'other' }
+      it { should create_deployment revision: github_revision }
+    end
 
-  describe 'given --revision_type s3 --bucket other --key bundle.zip' do
-    before { s3_revision[:s3Location].update(key: 'bundle.zip', bucket: 'other') }
-    it { should create_deployment revision: s3_revision }
+    describe 'given --revision_type s3 --bucket bucket --key bundle.zip' do
+      before { s3_revision[:s3Location][:key] = 'bundle.zip' }
+      it { should create_deployment revision: s3_revision }
+    end
+
+    describe 'given --revision_type s3 --bucket other --key bundle.zip' do
+      before { s3_revision[:s3Location].update(key: 'bundle.zip', bucket: 'other') }
+      it { should create_deployment revision: s3_revision }
+    end
+
+    describe 'given --revision_type s3 --bucket bucket --key other --bundle_type other' do
+      before { s3_revision[:s3Location].update(key: 'other', bundleType: 'other') }
+      it { should create_deployment revision: s3_revision }
+    end
+
+    describe 'given --wait_until_deployed' do
+      it { should create_deployment revision: github_revision }
+      it { should have_run '[print] Waiting for the deployment to finish ' }
+      it { should get_deployment }
+    end
   end
 
-  describe 'given --revision_type s3 --bucket bucket --key other --bundle_type other' do
-    before { s3_revision[:s3Location].update(key: 'other', bundleType: 'other') }
-    it { should create_deployment revision: s3_revision }
+  describe 'with ~/.aws/credentials' do
+    let(:args) { |e| %w(--application app) }
+
+    file '~/.aws/credentials', <<~str
+      [default]
+      aws_access_key_id=access_key_id
+      aws_secret_access_key=secret_access_key
+    str
+
+    before { subject.run }
+    it { should have_run '[info] Using Access Key: ac******************' }
   end
 
-  describe 'given --wait_until_deployed' do
-    it { should create_deployment revision: github_revision }
-    it { should have_run '[print] Waiting for the deployment to finish ' }
-    it { should get_deployment }
+  describe 'with ~/.aws/config' do
+    let(:args) { |e| %w(--access_key_id id --secret_access_key secret) }
+
+    file '~/.aws/config', <<~str
```

Comment:
`<<~` is not valid in Ruby 2.2.x, so we need a different here-doc marker.

Reply:
true. i keep forgetting that 😞

Reply:
done!